gematria/experiments/access_pattern_bm/contiguous_matrix.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/experiments/access_pattern_bm/contiguous_matrix.h"
+
+#include <immintrin.h>
+
+#include <memory>
+#include <random>
+
+#include "absl/base/optimization.h"
+
+namespace gematria {
+
+static std::default_random_engine generator;
+static std::uniform_int_distribution<int> distribution(0, 1023);
+
+std::unique_ptr<int[]> CreateRandomContiguousMatrix(const std::size_t size) {
+  auto matrix = std::make_unique<int[]>(size * size);
+
+  for (int i = 0; i < size; ++i) {
+    for (int j = 0; j < size; ++j) {
+      matrix[size * i + j] = distribution(generator);
+    }
+  }
+
+  return matrix;
+}
+
+void FlushContiguousMatrixFromCache(std::unique_ptr<int[]> &matrix,
```

Comment:
So, if I understand correctly, I should also similarly have all other `Flush...FromCache` functions take `const T*` for whatever `T` works for that particular benchmark since there is no ownership transfer going on there either, right?